MoltenVK/MoltenVK/GPUObjects/MVKVulkanAPIObject.mm

```diff
@@ -27,12 +27,31 @@
 VkResult MVKVulkanAPIObject::setDebugName(const char* pObjectName) {
 	if (pObjectName) {
 		[_debugName release];
-		_debugName = [[NSString alloc] initWithUTF8String: pObjectName];	// retained
+		@autoreleasepool {
+			_debugName = [[NSString alloc] initWithUTF8String: pObjectName];	// retained
+		}
 		propagateDebugName();
 	}
 	return VK_SUCCESS;
 }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wobjc-method-access"
+void MVKVulkanAPIObject::setMetalObjectLabel(id mtlObj, NSString* label) {
+	if ( !label ) { return; }
+
+	// If debug mode is enabled, append the object address.
+	// Since doing so might internally create other Metal objects, so catch them in an autorelease pool.
+	if (getMVKConfig().debugMode) {
+		@autoreleasepool {
+			[mtlObj setLabel: [label stringByAppendingFormat: @" (%p)", mtlObj]];
+		}
+	} else {
+		[mtlObj setLabel: label];
```

Comment:
Should this call also be wrapped in an autorelease pool? I fail to see how calling `-[NSString stringByAppendingFormat:]` would result in Metal creating more objects, though I understand that the string needs to be released once it's been set as the label. I _could_ see, however, `-setLabel:` triggering the creation of Metal objects, particularly with one of the debug Metal runtimes loaded.

Reply:
My intention is to avoid creating an autorelease pool unless absolutely necessary, since `setLabel:` can be called in some performance-sensitive areas.

I expect that, internally, `setLabel:` simply sets the label property (ivar) and sends `retain` to the label object. I don't see where it would be creating more objects by doing that. Can you elaborate on that?

With `stringByAppendingFormat:`, I expect a bunch of internal allocs to parse the variable input, convert it to strings, and build the larger string. If this work is done in C, not a problem, but if done in Obj-C, it would result in a bunch of temp Obj-C objects being created internally.